### DISCUSSION TITLE
Update tab state when loading a file.

### DISCUSSION THREAD

**Discussion by JipeB** [vlachoudis/bCNC#1838] [General]
Hello,
When I load a file intended for a workspace other than G54, the display does not update.
If anyone knows how to change this behavior, I'm all ears.
JP

**Comment by rschell** [comment]
First thought when loading file in CNC.py around statement 2422 where bCNC is loading file into editor, scan lines for G54-G59.  When found, change WCS state in ControlPage.py in the StateFrame class.  This might require binding a "WCSupdate" string to the procedure in bmain.py to trigger button update in ControlPage.py. 

What are the chances that there are multiple workspace changes in a file? If so, only process the first one?
Ron

**Comment by JipeB** [comment]
Thank you for this lead, I have already modified CNC.py, but I did not see the method to cause the update.
It's rare to change WCS several times in the same program (at least on small machines), identifying the initial parameter is sufficient as far as I'm concerned.
It would also be interesting to do the same update for the UNITS imperial/metric and PLANE XYZ toggle.
JP

**Reply by JipeB** [reply]
@rschell  Hi Ron,
You will find in the attached file the modifications made to obtain the automatic switching of the workspace when loading the GCode file. If the file contains several changes, only the first is taken into account (Part origin for the first machining); the other offsets (if any) will be taken into account during the execution of the program.
The G5... command must be alone on the line (this is the case in most programs established by CAM (Fusion360, FreeCad... and in accordance with industrial practices).
JP

[WcsAuto.pdf](https://github.com/vlachoudis/bCNC/files/12267548/WcsAuto.pdf)
